protostar/commands/test/cheatcodes/reflect_cheatcode.py

```diff
@@ -0,0 +1,224 @@
+                            ),
+                        )
+                    )
+
+                else:
+                    assert isinstance(expr_type, TypePointer) and isinstance(
+                        expr_type.pointee, TypeStruct
+                    ), "Type must be of the form T*."
+
+                    tmp = deepcopy(curr._context.memory[addr])  # type: ignore
+                    assert isinstance(tmp, RelocatableValue)
+                    curr_dict[name] = tmp
+
+        assert isinstance(value._struct_definition, StructDefinition)
+        return _ReflectTreeNode(
+            typename=value._struct_definition.full_name.path[1], value=value_tree
+        )
+
+    # pylint: disable=R0201
+    def _convert_to_named_tuple(self, tree: _ReflectTreeNode) -> REFLECT_RETURN_TYPE:
+        stack: List[REFLECT_VALUE_TYPE] = [tree]
+        pre_order: List[REFLECT_VALUE_TYPE] = []
+
+        while len(stack) > 0:
+            curr = stack.pop()
+            pre_order.append(curr)
+            if isinstance(curr, _ReflectTreeNode):
+                assert isinstance(curr.value, dict)
+                for _, node in curr.value.items():
+                    stack.append(node)
+
+        # When we reach each element it contains a single value or namedtuple instead of a dict
+        for node in reversed(pre_order):
+            if not isinstance(node, _ReflectTreeNode):
+                # We can't do anything here either way
+                continue
+
+            # Convert all of your children into named tuples
+            assert isinstance(node.value, dict)
+            for name, child in node.value.items():
+
+                if not isinstance(child, _ReflectTreeNode):
+                    node.value[name] = child
+                else:  # child.value is a Dict[str, NamedTuple]
+                    assert isinstance(child.value, dict)
+                    tpl = _get_pretty_tuple(
+                        child.typename,
+                        [(key, type(elem)) for key, elem in child.value.items()],
+                    )
+
+                    # pylint: disable=E1102
+                    node.value[name] = tpl(*child.value.values())
+
+        assert isinstance(tree.value, dict)
+        tpl = _get_pretty_tuple(
+            tree.typename, [(key, type(elem)) for key, elem in tree.value.items()]
+        )
+
+        # pylint: disable=E1102
+        return tpl(*tree.value.values())
+
+    def reflect(self, value: REFLECT_INPUT_TYPE) -> REFLECT_RETURN_TYPE:
+        tree = self._generate_value_tree(value)
+        if not isinstance(tree, _ReflectTreeNode):
+            return tree
+        tpl = self._convert_to_named_tuple(tree)
+        return tpl
+
+
+def _isinstance_namedtuple(obj: Any) -> bool:
+    return (
+        isinstance(obj, tuple) and hasattr(obj, "_asdict") and hasattr(obj, "_fields")
+    )
+
+
+def _get_pretty_tuple(name: str, tuple_args: List[Tuple[str, Type]]) -> Type:
```

Comment:
```suggestion
def PrettyNamedTuple(name: str, tuple_args: List[Tuple[str, Type]]) -> Type:
```
maybe? to be consistent with NamedTuple